src/stp/Play.cpp

```diff
@@ -42,15 +42,19 @@ void Play::update() noexcept {
 
     for (auto& role : roles) {
         // Update the roles
-        auto roleStatus = role->update(stpInfos[role->getName()]);
-        roleStatuses.emplace_back(roleStatus);
+        if (stpInfos.find(role->getName()) != stpInfos.end()) {
+            auto roleStatus = role->update(stpInfos[role->getName()]);
+            roleStatuses.emplace_back(roleStatus);
 
-        if (roleStatus == Status::Waiting) {
-            // Should role skip end tactic?
-            if (shouldRoleSkipEndTactic()) {
-                // TODO: force role to go to next tactic
-                // role.forceNextTactic(); (not implemented yet)
+            if (roleStatus == Status::Waiting) {
+                // Should role skip end tactic?
+                if (shouldRoleSkipEndTactic()) {
+                    // TODO: force role to go to next tactic
+                    // role.forceNextTactic(); (not implemented yet)
+                }
             }
+        } else {
+            RTT_ERROR("Trying to update role that is not in STPInfos!")
```

Comment:
I think I changed my mind, this should not be an error. It is an expected flow when robots drop out/break/shutdown so maybe make this debug?
```suggestion
            RTT_DEBUG("Trying to update role [", role->getName(), "] which is not in STPInfos!")
```